internal/cmd/kafka/command_link_create.go

```
@@ -167,36 +195,125 @@ func (c *linkCommand) create(cmd *cobra.Command, args []string) error {
+	"auto.create.mirror.topics.filters",
+	"cluster.link.prefix",
+	"topic.config.sync.ms",
+	"topic.config.sync.include",
+	"mirror.start.offset.spec",
+	"security.protocol",
+	"sasl.mechanism",
+	"local.security.protocol",
+	"local.sasl.mechanism")
+
+func linkConfigsCommandOutput(linkConfig map[string]string) string {
+	filtered := make(map[string]string)
+	for key, val := range linkConfig {
+		if printableLinkConfigs.Contains(key) {
+			filtered[key] = val
+		}
+	}
+	return properties.CreateKeyValuePairs(filtered)
+}
+
 func getJaasValue(apiKey, apiSecret string) string {
 	return fmt.Sprintf(`%s username="%s" password="%s";`, jaasConfigPrefix, apiKey, apiSecret)
 }
 
-func (c *linkCommand) getConfigMapAndLinkMode(configFile string) (map[string]string, linkMode, error) {
+func (c *linkCommand) getConfigMapAndLinkMode(configFile string) (map[string]string, *linkModeMetadata, error) {
 	if configFile != "" {
 		var linkMode linkMode
 		configMap, err := properties.FileToMap(configFile)
 		if err != nil {
-			return nil, linkMode, err
+			return nil, nil, err
 		}
 		linkModeStr, ok := configMap["link.mode"]
 		if !ok {
 			// Default is destination if no config value is provided.
 			linkMode = Destination
-		} else if strings.EqualFold(linkModeStr, "DESTINATION") {
-			linkMode = Destination
-		} else if strings.EqualFold(linkModeStr, "SOURCE") {
-			linkMode = Source
+			linkModeStr = DESTINATION
 		} else {
-			return nil, linkMode, errors.Errorf(`unrecognized link.mode "%s". Use DESTINATION or SOURCE.`, linkModeStr)
+			switch linkModeStr {
+			case DESTINATION:
+				linkMode = Destination
+			case SOURCE:
+				linkMode = Source
+			case BIDIRECTIONAL:
+				linkMode = Bidirectional
+			default:
+				return nil, &linkModeMetadata{linkMode, linkModeStr}, unrecognizedLinkModeErr(linkModeStr)
+			}
 		}
-		return configMap, linkMode, nil
+		configMap["link.mode"] = linkModeStr
+		return configMap, &linkModeMetadata{linkMode, linkModeStr}, nil
```

I feel like we should only have to return `linkMode` from this function, since each linkMode has an associated string that we can look up later.

---

I didn't find an easy way to associate a String with an enum/iota. Do you know of one? I didn't want to maintain a Map and have to deal with error checking if the lookup fails.

---

> I didn't want to maintain a Map and have to deal with error checking if the lookup fails.

This is the best way that I know to do it. You can just ignore any errors since the set of inputs are constrained to what's in the enum, and you should never get any other type.

---

Yeah that's def a fair option. My personal preference is to avoid having to worry about ignoring errors so will leave as is even if it is a bit clunky.